src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb

```diff
@@ -242,6 +241,48 @@ Namespace Microsoft.VisualStudio.Editors.AppDesCommon
             Return False
         End Function
 
+        ''' <summary>
+        ''' Logs the given exception and returns whether or not this is a "recoverable" exception, i.e. exception can be ignored.
+        ''' </summary>
+        ''' <param name="ex">The exception to log and check if it is recoverable.</param>
+        ''' <param name="exceptionEventDescription">Additional description for the cause of the exception.</param>
+        ''' <param name="throwingComponentName">Name of the component that threw that exception, generally the containing type name.</param>
+        ''' <param name="debugFail">If True, then invoke Debug.Fail for recoverable exception.</param>
+        ''' <param name="debugWriteLine">If True, then invoke Debug.WriteLine for recoverable exception.</param>
+        ''' <param name="considerExceptionAsRecoverable">If True, then the given exception is always considered recoverable.
+        ''' Otherwise, invokes <see cref="IsUnrecoverable(Exception, Boolean)"/> to determine if exception is recoverable or not.</param>
+        ''' <param name="ignoreOutOfMemory">If True, out of memory will be considered recoverable.</param>
+        ''' <remarks></remarks>
+        Public Function ReportWithoutCrash(ByVal ex As Exception,
```

Comment:
I'd approach this just slightly different - you've done a good job replicating all the different behaviors that they had, but I wouldn't do this - let's pick a policy for the entire project system, and just apply it consistently. ie Either we always call through IsUnrecoverable or we don't (I vote never), either we always Debug.Fail or we don't (I'd probably opt for always just so we can get a feel of how bad this is under debug).